torch/utils/_tensor_creation.py

```diff
@@ -0,0 +1,62 @@
+import torch
+
+from torch._torch_docs import factory_common_args
+
+
+def from_file(
+    filename,
+    shared=None,
+    size=0,
+    *,
+    dtype=None,
+    layout=None,
+    device=None,
+    pin_memory=False,
+):
+    r"""
+
+    Creates a CPU tensor with a storage backed by a memory-mapped file.
+
+    If ``shared`` is True, then memory is shared between processes. All changes are written to the file.
+    If ``shared`` is False, then changes to the tensor do not affect the file.
+
+    ``size`` is the number of elements in the Tensor. If ``shared`` is ``False``, then the file must contain
+    at least :math:`size * sizeof(dtype)` bytes. If ``shared`` is `True` the file will be created if needed.
+
+    .. note::
+        Only CPU tensors can be mapped to files.
+
+    .. note::
+        For now, tensors with storages backed by a memory-mapped file cannot be copied to pinned memory.
```

Comment:
nit: they can be copied to pinned memory right? They cannot be created in pinned memory here?